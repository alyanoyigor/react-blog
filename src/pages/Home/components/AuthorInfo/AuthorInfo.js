import React from 'react';
import AuthorImagePath from '../../../../assets/img/authorImage.jpeg';
import {
  StyledAuthorInfoContainer,
  StyledCircle,
  StyledImageContainer,
  StyledTitle,
} from './styled';

console.log(AuthorImagePath);

export const AuthorInfo = () => {
  return (
    <StyledAuthorInfoContainer>
      <StyledTitle>
        Hi, Iâ€™m Igor
        <p className="special-title">Frontend Developer</p>
      </StyledTitle>
      <StyledImageContainer>
        <img src={AuthorImagePath} alt="Igor Alyanoy" />
        <StyledCircle />
      </StyledImageContainer>
    </StyledAuthorInfoContainer>
  );
};
