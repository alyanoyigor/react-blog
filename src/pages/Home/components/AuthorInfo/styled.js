import styled from 'styled-components';

export const StyledTitle = styled.h1`
  font-size: 32px;
  font-weight: 700;
  letter-spacing: 1px;
  margin: 0;

  & .special-title {
    margin: 0;
    padding-top: 16px;
    font-size: 64px;
    background: radial-gradient(
      53.87% 5889.12% at 44.59% 58.82%,
      #bd3ff6 0,
      #66d9fb 100%
    );
    -webkit-background-clip: text;
    background-clip: text;
    color: transparent;
  }
`;

export const StyledImageContainer = styled.div`
  display: flex;
  position: relative;
  width: 300px;
  height: 300px;

  & img {
    z-index: 1;
    border-radius: 50%;
  }
`;

export const StyledCircle = styled.span`
  left: 16px;
  top: 16px;
  position: absolute;
  height: 100%;
  width: 100%;
  background: radial-gradient(
    53.87% 5889.12% at 44.59% 58.82%,
    #bd3ff6 0,
    #66d9fb 100%
  );
  border-radius: 50%;
  display: inline-block;
`;

export const StyledAuthorInfoContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
