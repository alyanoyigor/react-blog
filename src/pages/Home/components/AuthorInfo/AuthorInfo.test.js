import React from 'react';
import { render } from '@testing-library/react';
import { AuthorInfo } from './AuthorInfo';

describe('Author Info', () => {
  test('should render AuthorInfo', () => {
    const { container } = render(<AuthorInfo />);
    expect(container).toBeInTheDocument();
  });

  test('should match to snapshot', () => {
    const { container } = render(<AuthorInfo />);
    expect(container).toMatchSnapshot();
  });
});
