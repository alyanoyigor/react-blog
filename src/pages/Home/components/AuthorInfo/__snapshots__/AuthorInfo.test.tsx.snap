// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Info should match to snapshot 1`] = `
<div>
  <div
    class="sc-hKMtZM kLqRXc"
  >
    <h1
      class="sc-bczRLJ bmlJio"
    >
      Hi, Iâ€™m Igor
      <p
        class="special-title"
      >
        Frontend Developer
      </p>
    </h1>
    <div
      class="sc-gsnTZi kvsHCz"
    >
      <img
        alt="Igor Alyanoy"
        src="authorImage.jpeg"
      />
      <span
        class="sc-dkzDqf kejsfl"
      />
    </div>
  </div>
</div>
`;
